partition.h

```diff
@@ -19,13 +19,17 @@
 #ifndef LIBSEMIGROUPS_PARTITION_H_
 #define LIBSEMIGROUPS_PARTITION_H_
 
+#include <algorithm>
 #include <vector>
 
+using std::vector;
+
 namespace libsemigroups {
 
   // Non-abstract
   // Class for representing a partition of a semigroup, i.e. an equivalence
-  // relation on some subset of the semigroup.
+  // relation on some subset of the semigroup.  T is the type of element stored,
+  // and may be a pointer type, e.g. word_t*
   //
   template <class T> class Partition {
```

Comment:
I looked this up and I understand the two are equivalent.  But perhaps `typename` would be better, since it doesn't imply that we're using a class (e.g. in the main use-case, `Partition<word_t*>`)